core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1033,10 +1086,22 @@ public long createFile(AlluxioURI path, CreateFileOptions options)
       BlockInfoException, IOException, FileDoesNotExistException {
     Metrics.CREATE_FILES_OPS.inc();
     try (JournalContext journalContext = createJournalContext();
-        LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE)) {
-      mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
+        LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE);
+        MasterAuditContext auditContext = createAuditContext("createFile", path, null, null)) {
+      if (options.isRecursive()) {
```

Comment:
What are the semantics (what should be logged) for an audit log message for a create operation?

Why is recursive create special?